src/main/java/com/dougnoel/sentinel/elements/Element.java

```diff
@@ -695,6 +703,69 @@ public Element hover() {
 	public String getTooltipText() {
 		hover();
 		return driver().findElement(By.xpath("//*[contains(text(),'')]")).getText();
-	}	
+	}
+	
+	/**
+	 * Gets the most prevalent color present in the element.
+	 * 
+	 * @return A color type of the most common color on the element.
+	 */
+	public Color getMostPrevalentColor() throws IOException {
+		byte[] imageAsByteArray = element().getScreenshotAs(OutputType.BYTES);
+		ByteArrayInputStream imageByteStream = new ByteArrayInputStream(imageAsByteArray);
+
+		BufferedImage elementImage = ImageIO.read(imageByteStream);
+		BufferedImage scaledImage = (BufferedImage) elementImage.getScaledInstance(1, 1, Image.SCALE_REPLICATE);
+		var rgb = scaledImage.getRGB(0, 0);
+		return new Color(rgb);
+	}
+	
+	/**
+	 * Takes a screenshot of the current element.
+	 * 
+	 * @return A screenshot of the current element as a File output type.
+	 */
+	public File getScreenshot() {
+		return element().getScreenshotAs(OutputType.FILE);
+	}
+	
+	/**
+	 * This method is used to get the background color of the element, or it's inherited background color.
+	 * 
+	 * @return The background color of the element converted to a java color object.
+	 */
+	public Color getBackgroundColor()
+	{  
+		return ascendTransparentColorElements(element());
+	}
 	
+	/**
+	 * Returns the color of the background behind the element.
+	 * 
+	 * @param element The web element to use as the root for checking for background colors.
+	 * 
+	 * @return The background color of the first parent with one or white if no inherited background color is found.
+	 */
+	private Color ascendTransparentColorElements(WebElement element) {
+		Color transparent = org.openqa.selenium.support.Colors.TRANSPARENT.getColorValue().getColor();
+		Color currentColor = org.openqa.selenium.support.Color.fromString(element.getCssValue("background-color")).getColor();
+		WebElement parentElement = null;
+				
+		if(currentColor.equals(transparent)) {
+			try {
+				parentElement = element.findElement(By.xpath("./.."));
+			} catch(NoSuchElementException e) {
+				//Suppress error when there are no more parents
+			}
+			
+			if(parentElement != null) {
+				return(ascendTransparentColorElements(parentElement));
+			}
+			else {
+				return Color.white;
+			}
+		} else {
+			return currentColor;
+		}
+	}
```

Comment:
The logic of this is a bit convoluted. Look up Big O notation for an in-depth discussion of the math, but basically for a self-recursive method we want as little branching and if statements as possible. We can reduce that by getting rid of branches. First we can get rid of the last else.
```
		Color currentColor = org.openqa.selenium.support.Color.fromString(element.getCssValue("background-color")).getColor();
		WebElement parentElement = null;

		if(currentColor.equals(transparent)) {
			try {
				parentElement = element.findElement(By.xpath("./.."));
			} catch(NoSuchElementException e) {
				//Suppress error when there are no more parents
			}
			
			if(parentElement != null) {
				return(ascendTransparentColorElements(parentElement));
			}
			else {
				return Color.white;
			}
		}
		return currentColor;
```

Now if we look at the logic in your recursion, we stop when there are no more parents - when we get a NoSuchElementException at that point we test farther below and return white. But as soon as that exception is thrown, we know we can return that color. By doing it there, we remove the need for another branch. It also solves the problem that you should never suppress exceptions. They should always be handled or thrown.

```
		Color currentColor = org.openqa.selenium.support.Color.fromString(element.getCssValue("background-color")).getColor();
		WebElement parentElement = null;

		if(currentColor.equals(transparent)) {
			try {
				parentElement = element.findElement(By.xpath("./.."));
			} catch(NoSuchElementException e) {
				return Color.white;
			}

			if(parentElement != null) {
				return(ascendTransparentColorElements(parentElement));
			}
		}
		return currentColor;
```

Our cognitive complexity is reducing, making the code easier to read, and that is also making the code execute faster because less code is being compiled in our recursive function. But we can reduce it further. We are doing a null test because we want to make sure parentElement is valid. But, because we are returning if it isn't, we no longer need that chack. We can take that branch out entirely. While we are at it, the return statement doesn't need parentheses around it so we should fix that.

```
		Color currentColor = org.openqa.selenium.support.Color.fromString(element.getCssValue("background-color")).getColor();
		WebElement parentElement = null;

		if(currentColor.equals(transparent)) {
			try {
				parentElement = element.findElement(By.xpath("./.."));
			} catch(NoSuchElementException e) {
				return Color.white;
			}

			return ascendTransparentColorElements(parentElement);
		}
		return currentColor;
```

Now let's talk about memory usage. parentElement is only used on those two lines of code. We can collapse them into one line to avoid using the variable at all and save some memory. While we are at it I'm going to rename the method to match the comment I made on line 749. If the find method execution fails it will throw the exception and the method will never be called recursively, instead, it will return white.

```
		Color currentColor = org.openqa.selenium.support.Color.fromString(element.getCssValue("background-color")).getColor();

		if(currentColor.equals(transparent)) {
			try {
				return getBackgroundColor(element.findElement(By.xpath("./.."));
			} catch(NoSuchElementException e) {
				return Color.white;
			}
		}
		return currentColor;
```

We've gone from 19 lines of code to 10. This is pretty good, but we can still optimize the speed. We still have an implicit if-else and so we can increase execution speed by returning the positive value first, so that we do not have to evaluate the rest of the code. As soon as we know what we are going to return, we return it.

```suggestion
		Color currentColor = org.openqa.selenium.support.Color.fromString(element.getCssValue("background-color")).getColor();

		if(!currentColor.equals(transparent))
			return currentColor;
		try {
			return getBackgroundColor(element.findElement(By.xpath("./.."));
		} catch(NoSuchElementException e) {
			return Color.white;
		}
```

We've now gotten it done to 9 lines from 19, eliminated a variable, made the execution faster, and gotten rid of a number of branching statements. (This also makes the unit test easier.)